x-pack/solutions/security/plugins/elastic_assistant/server/ai_assistant_data_clients/knowledge_base/index.ts

```diff
@@ -156,7 +151,8 @@ export class AIAssistantKnowledgeBaseDataClient extends AIAssistantDataClient {
   };
 
   public getInferenceEndpointId = async () => {
-    if (!this.options.assistantDefaultInferenceEndpoint) {
+    const elserId = await this.options.getElserId();
+    if (!this.options.assistantDefaultInferenceEndpoint || elserId !== ELSER_MODEL_2) {
```

Comment:
Why the `elserId !== ELSER_MODEL_2` check here for falling back to the security elser inference endpoint?

Reply:
for `pt_tiny_elser` for testing

Reply:
Okie, just checking as `getElserId()` will return `.elser_model_2_linux-x86_64` instead of `.elser_model_2` on cloud deployments.

Reply:
great catch @spong